File: 2025/day01/Prob2.java
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class Prob2 {

    public static void main(String args[]) throws Exception {
        List<String> rotations = Files.readAllLines(Path.of("2025/day01/input.txt"));
        int position = 50;
        int n = 0;
        for (String rotation: rotations) {
            char direction = rotation.charAt(0);
            int distance = Integer.parseInt(rotation.substring(1));
            int oldPosition = position;
            if (direction == 'L') {
                position -= distance;
                if (position < 0) {
                    n += (Math.abs(position))/100 + (oldPosition == 0 ? 0 : 1);
                } else if (position == 0) {
                    n++;
                }
            } else {
                position += distance;
                n += position/100;
            }
            position = mod(position, 100);
        }
        System.out.println(n);
    }

    public static int mod(int n, int m) {
        while (n < 0){
            n += m;
        }
        while (n >= m) {
            n -= m;
        }
        return n;
    }
    
}

File: 2025/day01/prob2.py
def main():
    with open("2025/day01/input.txt") as f:
        rotations = f.readlines()
    position = 50
    n = 0
    for rotation in rotations:
        direction = rotation[0]
        distance = int(rotation[1:])
        oldPosition = position
        if direction == 'L':
            position -= distance
            if position < 0:
                n += abs(position)//100 + (0 if oldPosition == 0 else 1)
            elif position == 0:
                n += 1
        else:
            position += distance
            n += position//100
        position %= 100
    print(n)


if __name__ == "__main__":
    main()